lib/core/data/repositories/position_repository_impl.dart

```diff
@@ -0,0 +1,38 @@
+import 'package:geolocator/geolocator.dart' as geolocator;
+import 'package:injectable/injectable.dart';
+import 'package:movna/core/data/datasources/position_source.dart';
+import 'package:movna/core/domain/entities/position.dart' as domain;
+import 'package:movna/core/domain/repositories/position_repository.dart';
+
+@Injectable(as: PositionRepository)
+class PositionRepositoryImpl implements PositionRepository {
+  PositionRepositoryImpl({required this.positionSource});
+  PositionSource positionSource;
+
+  static domain.Position _positionGeolocatorToDomain(geolocator.Position pos) =>
+      domain.Position(
```

Comment:
là dessus j'aime bien utiliser le principe d'adapter: une classe dont le rôle est de transformer 1 modèle en 1 entité
sur le principe tu as une classe qui définit le comportement des adapter:
```dart
abstract class BaseAdapter<Entity, Model> {
  Entity modelToEntity(Model m) {
    // Method is defined so that subclasses don't have to implement it if they don't need it 
    throw UnimplementedError();
  }

  Model entityToModel(Entity e) {
    // Method is defined so that subclasses don't have to implement it if they don't need it 
    throw UnimplementedError();
  }
}
```

et ses implémentations, pour position ce serait

```dart
@injectable
class PositionAdapter extends BaseAdapter<domain.Position, geolocator.Position> {
  @override
  domain.Position modelToEntity(geolocator.Position model) {
    // Ptite tambouille ici
  }
}
```

et comme ça dans ton repository tu lui mets un attribut de type: `final PositionAdapter _positionAdapter`
et pour convertir `_positionAdapter.modelToEntity(pos)`

ça a l'avantage aussi de pouvoir définir des comportements un peu plus complexes dans `BaseAdapter` avec notamment de la conversion de lists etc...